tests/models/test_taskinstance.py

```diff
@@ -5312,3 +5312,65 @@ def test_swallow_mini_scheduler_exceptions(_schedule_downstream_mock, create_tas
     ti.schedule_downstream_tasks()
     assert "Error scheduling downstream tasks." in caplog.text
     assert "To be swallowed" in caplog.text
+
+
+def test_ti_selector_condition(dag_maker):
+    from airflow.utils.timezone import datetime
+
+    clear_db_runs()
+    files = ["a", "b", "c"]
+
+    start_date = datetime(2024, 1, 1)
+    files = ["file1", "file2", "file3"]
+
+    with dag_maker(dag_id="task_group_mapping_example", start_date=start_date, schedule=None, catchup=False):
+
+        @task_group(group_id="etl")
+        def etl_pipeline(file):
+            e = EmptyOperator(task_id="e")
+            t = EmptyOperator(task_id="t")
+            last = EmptyOperator(task_id="last")
+
+            e >> t >> last
+
+        etl_pipeline.expand(file=files)
+
+    dag_instance = dag_maker.dag
+    dag_maker.create_dagrun(
+        run_id="manual_run_2024_01_01",
+        state=State.SUCCESS,
+        execution_date=start_date,
+        start_date=start_date,
+        data_interval=(start_date, start_date),
+        run_type=DagRunType.SCHEDULED,
+    )
```

Comment:
Manual or scheduled? (It doesn’t really matter in this test but it’s just weird.)